src/ucs/stats/stats.c

```diff
@@ -150,6 +206,98 @@ static ucs_status_t ucs_stats_node_new(ucs_stats_class_t *cls, ucs_stats_node_t
     return UCS_OK;
 }
 
+static ucs_status_t ucs_stats_filter_node_new(ucs_stats_class_t *cls, ucs_stats_filter_node_t **p_node)
+{
+    ucs_stats_filter_node_t *node;
+
+    node = ucs_malloc(sizeof(ucs_stats_filter_node_t),
+                      "stats filter node");
+    if (node == NULL) {
+        ucs_error("Failed to allocate stats filter node for %s", cls->name);
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    *p_node = node;
+    return UCS_OK;
+}
+
+static ucs_stats_filter_node_t * ucs_stats_find_class(ucs_stats_filter_node_t *filter_parent,
+                                                      const char *class_name) {
+    ucs_stats_filter_node_t *filter_node;
+    ucs_stats_node_t * node;
+    ucs_list_for_each(filter_node, &filter_parent->children, list) {
+        node = ucs_list_head(&filter_node->type_list_head,
+                             ucs_stats_node_t,
+                             type_list);
+        if (!node) {
+            ucs_error("no list for node filter");
+            return NULL;
+        }
+        if (!strcmp(node->cls->name, class_name)) {
+            return filter_node;
+        }
+    }
+    return NULL;
+}
+
+static void ucs_stats_add_wildcard(ucs_stats_node_t *node,
```

Comment:
looks like ucs_stats_add_wildcard is not good name, maybe ucs_stats_add_to_filter?